src/merkle/store/tests.rs

```diff
@@ -647,6 +647,70 @@ fn test_constructors() -> Result<(), MerkleError> {
     Ok(())
 }
 
+#[test]
+fn node_path_should_be_truncated_by_midtier_insert() {
+    let key = 0b11010010_11001100_11001100_11001100_11001100_11001100_11001100_11001100_u64;
+
+    let mut store = MerkleStore::new();
+    let root: Word = EmptySubtreeRoots::empty_hashes(64)[0].into();
+
+    // insert first node - works as expected
+    let depth = 64;
+    let node = [Felt::new(key); WORD_SIZE];
+    let index = NodeIndex::new(depth, key).unwrap();
+    let root = store.set_node(root, index, node).unwrap().root;
+    let result = store.get_node(root, index).unwrap();
+    let path = store.get_path(root, index).unwrap().path;
+    assert_eq!(node, result);
+    assert_eq!(path.depth(), depth);
+    assert!(path.verify(index.value(), result, &root));
+
+    // flip the first bit of the key and insert the second node on a different depth
+    let key = key ^ (1 << 63);
+    let key = key >> 8;
+    let depth = 56;
+    let node = [Felt::new(key); WORD_SIZE];
+    let index = NodeIndex::new(depth, key).unwrap();
+    let root = store.set_node(root, index, node).unwrap().root;
+    let result = store.get_node(root, index).unwrap();
+    let path = store.get_path(root, index).unwrap().path;
+    assert_eq!(node, result);
+    assert_eq!(path.depth(), depth);
+    assert!(path.verify(index.value(), result, &root));
+
+    // attempt to fetch a path of the second node to depth 64
+    // should fail because the previously inserted node will remove its sub-tree from the set
+    let key = key << 8;
+    let index = NodeIndex::new(64, key).unwrap();
+    assert!(store.get_node(root, index).is_err());
+}
+
+#[test]
+fn get_leaf_depth_works() {
+    let key = u64::MAX;
```

Comment:
I think this test is a bit difficult to interpret. I don't mind having it, but I'd prefer to have something simpler as well. Could be something like this:

1. Initialize `MerkleStore` with some small SMT (i.e., depth 8 with 4 leaves).
2. Call `get_leaf_depth()` on one of the leaf nodes - make sure it works.
3. Call `get_leaf_depth()` on a couple empty nodes (ideally at different depths) - make sure it works as expected.

For the above, I'd use hard-coded values.

Reply:
Agreed, let's keep this test discussion centralized here

https://github.com/0xPolygonMiden/crypto/pull/125#discussion_r1158687618

Reply:
Added `get_leaf_depth_works_with_small_depth` with some 8 depth random keys and hardcoded cases